### ISSUE TITLE
Browser console warning when beginning checkout

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#3972]
## Issue Description
Receive the following warning in the browser console when clicking on the "checkout" button from the PDP. Also a spinner is shown for a beat or two. Seems to have no effect on actual functionality.

```
modules.js?hash=bf8f109c7b1f6e58e6c6a76de0bab6455bce22e4:64007 Warning: Can only update a mounted or mounting component. This usually means you called setState, replaceState, or forceUpdate on an unmounted component. This is a no-op.

Please check the code for the ProductDetailContainer component.
```

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. As an admin navigate to a product page
1. Click on "Add to Cart"
1. Click on the "checkout" button that appears briefly
3. Observe the spinner and the warning in the console

### Versions
Node: 8.9.4
NPM: 5.6.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.28.0
Reaction: 1.9.0
Reaction branch: release-1.9.0
Docker: 17.12.0-ce

**Comment by zenweasel** [comment]
This is an `impact-major` because console error/warnings are always at least majors.

**Comment by spencern** [comment]
I would say that console errors should be major if they impact core functionality, not sure if we should consider warnings major or minor, but let's knock this one out.

**Comment by Akarshit** [comment]
@prinzdezibel I am not able to reproduce this.

**Comment by prinzdezibel** [comment]
Was able to reproduce it after fiddling with setTimeout times. Providing a fix.

**Comment by zenweasel** [comment]
Closed via #3980
